flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/OperatorCodeGenerator.scala

```diff
@@ -109,10 +109,16 @@ object OperatorCodeGenerator extends Logging {
 
         $endInput
 
+        @Override
+        public void finish() throws Exception {
+            ${ctx.reuseFinishCode()}
+            super.finish();
```

Comment:
NIT: why `reuseFinishCode` first, `super.finish` second.

Reply:
similar to the close sequence, it is relatively more safer (not always strictly) to handle the actions of the child class first and call the parent public logic last. Btw, I changed the current close call at first but reverted it for focusing on this pr itself.